test/integration/performance_expansion_int_test.go

```diff
@@ -0,0 +1,329 @@
+package integration
+
+import (
+	"fmt"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ActiveState/cli/internal/config"
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/exeutils"
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/ActiveState/cli/internal/testhelpers/e2e"
+	"github.com/ActiveState/cli/internal/testhelpers/tagsuite"
+	"github.com/ActiveState/cli/pkg/projectfile"
+	"github.com/stretchr/testify/suite"
+	"gopkg.in/yaml.v2"
+)
+
+// Configuration values for the performance tests
+const (
+	DefaultProject = "https://platform.activestate.com/ActiveState-CLI/Yaml-Test/?branch=main&commitID=0476ac66-007c-4da7-8922-d6ea9b284fae"
+
+	DefaultMaxTime        = 1000 * time.Millisecond
+	DefaultSamples        = 10
+	DefaultVariance       = 0.75
+	DefaultSecretsMaxTime = 2100 * time.Millisecond
+	// Add other configuration values on per-test basis if needed
+)
+
+type PerformanceExpansionIntegrationTestSuite struct {
+	tagsuite.Suite
+}
+
+func (suite *PerformanceExpansionIntegrationTestSuite) startSvc(ts *e2e.Session) {
+	// Start svc first, as we don't want to measure svc startup time which would only happen the very first invocation
+	stdout, stderr, err := exeutils.ExecSimple(ts.SvcExe, []string{"start"}, []string{})
+	suite.Require().NoError(err, fmt.Sprintf("Full error:\n%v\nstdout:\n%s\nstderr:\n%s", errs.JoinMessage(err), stdout, stderr))
+}
+
+func (suite *PerformanceExpansionIntegrationTestSuite) TestExpansionPerformance() {
+	suite.OnlyRunForTags(tagsuite.Performance)
+	baseline := DefaultMaxTime
+	suite.Run("CallScript", func() {
+		avg := suite.testScriptPerformance(scriptPerformanceOptions{
+			script: projectfile.Script{
+				Name:     "call-script",
+				Value:    `echo "Hello World"`,
+				Language: "bash",
+			},
+			expect:  "Hello World",
+			samples: DefaultSamples,
+			max:     DefaultMaxTime,
+		})
+		variance := float64(avg) + (float64(avg) * DefaultVariance)
+		baseline = time.Duration(variance)
+	})
+
```

Comment:
Can we add a `suite.Require().NotEqual(baseline, DefaultMaxTime)` here to ensure that the baseline has been overwritten?

I feel like I've ran into some issues with `suite.Run()` being ran concurrently. Certainly the underlying implementation could be doing this. I don't think it's intended to be used in such a way that multiple `suite.Run()` calls impact each other.

Reply:
Good to know!